### ISSUE TITLE
Save game settings to GameData rather than SharedPreferences

### ISSUE THREAD

**Issue by BonzaiThePenguin** [mupen64plus-ae/mupen64plus-ae#423]
In the middle of moving the PlayMenuActivity logic over to the game sidebar, I remembered that the game settings are saved to SharedPreferences. Any reason for that? Seems like that would make it significantly harder for users to move their settings to a new device, and since SharedPreference is a single object it ends up having to generate a ton of unique keys to store everything, which just seems like a nasty hack to me.

Am I correct in thinking that saving to files in the GameData folder would be a better solution?

**Comment by littleguy77** [comment]
Yeah we might be headed that direction.  Many moons ago in version 1.x, preferences were stored in a  config file.  In late 2012 I joined the project and re-implemented the entire front-end for 2.x, using PreferenceActivity and SharedPreferences.  Those classes/concepts were designed for each other, and I liked that we could layout both the visual and logical structure of the menus/backing preferences in a single xml file.  It made it trivial to add new settings (which we were doing a lot of at the time): just add a few lines of xml in preferences.xml, add a final field in UserPrefs.java, and define the text in strings.xml.

With the heavy game-centric settings of version 3.x, and more importantly, the phase out of PreferenceActivity for UI, this approach is losing its luster.  I'm up for reconsidering the backend data structures for version 3.

I plan to get all your visual machinery changes into master this weekend.  That would be the material design styling, navigation drawers, appcompat-based implementation, etc.  From there, I need to follow a careful path to integrating your other changes:
 - strings.xml needs to be modified carefully, since this affects dozens of translators.  Sometimes innocuous changes add a lot of work for them.
 - Changes to the refresh library implementation.  I'm getting hard crashes on my device, and would like to understand the functional features/changes you made.

I'd be much happier if we could wait for the dust to settle before trying to change the data structure/implementation.  There are some subtleties (I learned the hard way in version 2) and it's nice if these kinds of changes can be made serially rather than in parallel to all the other changes.

I really love all your new ideas and how you implemented them.  I don't want to dampen your enthusiasm or slow you down, but I have learned (often the hard way) that with such major overhauls, it is essential that we have a clean git history.  Regressions are inevitable, but with a clean history it is usually very easy to isolate them.

Of course if you want to test all this on a brand-new branch off of master, that is perfectly fine too.  The difficulty is when one feature branch starts conflating many new features and it becomes difficult to disentangle the commits.
